---
title: "MOFA"
output: html_notebook
---



1) get data
2) Normalization
3) PCA, sample-sample correlation → determine outliers
4) remove low counts threshold → graph normalized counts against % of genes above threshold
5) Clustering based on transcriptomics and metabolomics (SNF)
6) DGE between clusters / clinical characteristics
7) Clustering genes → gene set enrichment analysis
8) Digital cell quantification (EPIC)
9) Transcriptomics aging clock (transcriptomics age, biological age)
10) MOFA → metabolomics and transcriptomics
11) take MOFA HIV data to validate our findings


### clean environment
```{r}
rm(list=ls())
```

### set directory
```{r setup}
    knitr::opts_knit$set(root.dir = normalizePath("~/Desktop/Code-PHD/COCOMO_txn/")) 
```

### load packages
```{r}
library(DESeq2)
library(dplyr)
library(SNFtool)
library(xlsx)
library(ggplot2)
library(corrr)
library(mygene)
library(org.Hs.eg.db)
library(edgeR)
```

## add genes to table
```{r}
dge <-read.csv("/home/flomik/Desktop/Code-PHD/COCOMO_txn/results/Deseq2/model3_results_DGE_2_1.csv", row.names = 1)
dge$Gene_name <- mapIds(org.Hs.eg.db, rownames(dge), keytype="ENSEMBL", column="SYMBOL")
dge <- dge[,c(7, 8, 2, 5, 6)]
write.xlsx(dge, "manuscript/table_transcriptomics_DGE_not_filt_model_3.xlsx")
```

```{r}
col <- c( "#005900", "#878782",	"#761212")
```

### transcriptomics
```{r}
data_txn <- read.csv("processing/COCOMO_transcriptomics_vsd_data_deseq2.csv", row.names = 1, check.names = FALSE)
#data_txn$COCOMO_ID <- rownames(data_txn)
```
### metabolomics
```{r}
data_met <- read.csv("processing/COCOMO_metabolomics_log2_norm_filt.csv", row.names = 1, check.names = FALSE)
data_met <- data_met[rownames(data_met) %in% rownames(data_txn), ]
data_met <- data_met[order(match(rownames(data_met), rownames(data_txn))),]
#data_met$COCOMO_ID <- rownames(data_met)
```

### olink
```{r}
data_olink <- read.csv("processing/olink_data_new_ids.csv", row.names = 1, check.names = FALSE)
data_olink$Condition <- NULL
```

/home/flomik/Desktop/Code-PHD/COCOMO_txn/data/lipidomics/clp_um_new_data.csv

### clinical parameters
```{r}
clinical <- read.csv("processing/clinical_data_clean_with_clusters_and_categories.csv")
```


```{r}
data_txn <- data_txn[rownames(data_txn) %in% rownames(data_olink),]
data_met <- data_met[rownames(data_met) %in% rownames(data_olink),]
clinical <- clinical[clinical$COCOMO_ID %in% rownames(data_olink), ]
```

```{r}
data_met <- data_met[order(match(rownames(data_met), rownames(data_txn))),]
data_olink <- data_olink[order(match(rownames(data_olink), rownames(data_txn))),]
```

```{r}
data_olink$COCOMO_ID <- rownames(data_olink)
data_met$COCOMO_ID <- rownames(data_met)
data_txn$COCOMO_ID <- rownames(data_txn)
```

## PCA Olink
```{r}
clinical_2 <- clinical[,c(1:2)]
data_olink$COCOMO_ID <- rownames(data_olink)
data <- merge(clinical_2, data_olink, by = "COCOMO_ID")

x <- read.csv("processing/bblearn_find_driver_genes_proteomics.csv")
x <- x[x$BIC_diff > 0,]
data <- data[,colnames(data) %in% c("cluster", x$gene)]
data <- data[data$cluster != "Ctrl",]
```

```{r}
pca <- prcomp(data[,-c(1)], scale. = FALSE)
df_out <- as.data.frame(pca$x)
df_out$group <- data$cluster
x <- project.pca.proportionvariances <- ((pca$sdev^2) / (sum(pca$sdev^2)))*100
a1 <- x[1]
a2 <- x[2]
 
gg <- data.frame(cluster=factor(df_out$group), x=df_out$PC1, y=df_out$PC2)
centroids <- aggregate(cbind(x,y)~cluster,data=gg,mean)
gg <- merge(gg,centroids,by="cluster",suffixes=c("",".centroid"))

ggplot(gg)+
      geom_point(aes(x=x,y=y,color=cluster, fill = cluster), size=3)+
      stat_ellipse(geom = "polygon", aes(x=x,y=y,color=cluster, fill = cluster), alpha = 0.3) +
      geom_point(data=centroids, aes(x=x, y=y, color=cluster, fill = cluster), size = 5, alpha = 0.9, shape = 19)+ 
      geom_segment(aes(x=x.centroid, y=y.centroid, xend=x, yend=y, color=cluster))+
      theme(legend.title=element_text(size=15),legend.text=element_text(size=12),
            legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
            axis.title.y=element_text(size=15),axis.title.x=element_text(size=15),
            axis.text.y=element_text(size=18),axis.text.x=element_text(size=18))+
      xlab(paste0("PC1: ",round(a1,0),"% variance"))+
      ylab(paste0("PC2: ",round(a2,0),"% variance"))+ 
        theme(axis.title.x = element_text(size=16, face = "bold",color="black"),
              axis.title.y = element_text(size=16, face = "bold",color="black"),
              axis.text = element_text(color="black",size=16, face = "bold"),
              axis.text.x =element_text(color="black",size=16, face = "bold"),
              panel.border = element_blank(),
              axis.ticks.x = element_blank(),
              strip.text.x = element_text(size = 9,color="black"),
              panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
              panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
              panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
              plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
              legend.background=element_rect(fill = alpha("white", 0)))+ scale_color_manual(values = col[-1])+ scale_fill_manual(values = col[-1])

path_fig <- paste0("results/figures/PCA/PCA_proteomics_biomarkers.pdf")
ggsave(path_fig, width = 7, height = 6)

ggplot(gg)+
      geom_point(aes(x=x,y=y,color=cluster, fill = cluster), size=3)+
      stat_ellipse(geom = "polygon", aes(x=x,y=y,color=cluster, fill = cluster), alpha = 0.3) +
      geom_point(data=centroids, aes(x=x, y=y, color=cluster, fill = cluster), size = 5, alpha = 0.9, shape = 19)+ 
      geom_segment(aes(x=x.centroid, y=y.centroid, xend=x, yend=y, color=cluster))+
      theme(legend.title=element_text(size=15),legend.text=element_text(size=12),
            legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
            axis.title.y=element_text(size=15),axis.title.x=element_text(size=15),
            axis.text.y=element_text(size=18),axis.text.x=element_text(size=18))+
      xlab(paste0("PC1: ",round(a1,0),"% variance"))+
      ylab(paste0("PC2: ",round(a2,0),"% variance"))+ 
        theme(axis.title.x = element_text(size=16, face = "bold",color="black"),
              axis.title.y = element_text(size=16, face = "bold",color="black"),
              axis.text = element_text(color="black",size=16, face = "bold"),
              axis.text.x =element_text(color="black",size=16, face = "bold"),
              panel.border = element_blank(),
              axis.ticks.x = element_blank(),
              strip.text.x = element_text(size = 9,color="black"),
              panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
              panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
              panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
              plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
              legend.background=element_rect(fill = alpha("white", 0)))+ scale_color_manual(values = col[-1])+ scale_fill_manual(values = col[-1])

path_fig <- paste0("results/figures/PCA/PCA_proteomics_biomarkers_V2.pdf")
ggsave(path_fig, width = 7, height = 3)
```

```{r}
fac <- read.csv("processing/factor_differing_groups_3.csv")
fac <- fac$x
```

```{r}
clinical[is.na(clinical)] <- "NA"
```
      
      
```{r}
clinical <- clinical[colnames(clinical) %in% c( "COCOMO_ID", "cluster", fac)]
```



```{r}
pca_data <- merge(clinical, data_txn, by = "COCOMO_ID", all.y = TRUE)
```

```{r}
col <- c("#878782",	"#761212")
```

```{r}
pca_data <- pca_data[pca_data$cluster != "Ctrl",]
```

```{r}
pca <- prcomp(pca_data[,-c(1,2)], scale. = FALSE)
df_out <- as.data.frame(pca$x)
df_out$group <- pca_data$cluster
x <- project.pca.proportionvariances <- ((pca$sdev^2) / (sum(pca$sdev^2)))*100
a1 <- x[1]
a2 <- x[2]
 
gg <- data.frame(cluster=factor(df_out$group), x=df_out$PC1, y=df_out$PC2)
centroids <- aggregate(cbind(x,y)~cluster,data=gg,mean)
gg <- merge(gg,centroids,by="cluster",suffixes=c("",".centroid"))

ggplot(gg)+
      geom_point(aes(x=x,y=y,color=cluster, fill = cluster), size=3)+
      stat_ellipse(geom = "polygon", aes(x=x,y=y,color=cluster, fill = cluster), alpha = 0.3) +
      geom_point(data=centroids, aes(x=x, y=y, color=cluster, fill = cluster), size = 5, alpha = 0.9, shape = 19)+ 
      geom_segment(aes(x=x.centroid, y=y.centroid, xend=x, yend=y, color=cluster))+
      theme(legend.title=element_text(size=15),legend.text=element_text(size=12),
            legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
            axis.title.y=element_text(size=15),axis.title.x=element_text(size=15),
            axis.text.y=element_text(size=18),axis.text.x=element_text(size=18))+
      xlab(paste0("PC1: ",round(a1,0),"% variance"))+
      ylab(paste0("PC2: ",round(a2,0),"% variance"))+ 
        theme(axis.title.x = element_text(size=16, face = "bold",color="black"),
              axis.title.y = element_text(size=16, face = "bold",color="black"),
              axis.text = element_text(color="black",size=16, face = "bold"),
              axis.text.x =element_text(color="black",size=16, face = "bold"),
              panel.border = element_blank(),
              axis.ticks.x = element_blank(),
              strip.text.x = element_text(size = 9,color="black"),
              panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
              panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
              panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
              plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
              legend.background=element_rect(fill = alpha("white", 0)))+ scale_color_manual(values = col)+ scale_fill_manual(values = col)

path_fig <- paste0("results/figures/PCA/PCA_transcriptomics.pdf")
ggsave(path_fig)
```

## make pca
```{r}
#' Make PCA plot for different parameters
#'
#' @param data dataframe containing features as columns and samples as row
#' @param clinical_data dataframe containing clinical information as column and samples as row
#' @param y string name of feature to analyse
#' @param centroids string add centroids or not ("yes"/"no")
#' @param scaling scale data (TRUE/FALSE)
#' @param colors list of colors
#' @param type string type of data
#' @param analysis string analysis
my_pca <- function(data, clinical_data, y, centroids, scaling, colors, type, analysis, label){
  pca_data <- merge(clinical_data, data, by = "COCOMO_ID", all.y = TRUE)
  pca <- prcomp(pca_data[,-c(1:ncol(clinical_data))], scale. = scaling)
  df_out <- as.data.frame(pca$x)
  df_out$group <- as.factor(pca_data[,colnames(pca_data) == y])
  x <- project.pca.proportionvariances <- ((pca$sdev^2) / (sum(pca$sdev^2)))*100
  a1 <- x[1]
  a2 <- x[2]
  if (colors == "yes") {
    if (centroids == "yes") {
      gg <- data.frame(cluster=factor(df_out$group), x=df_out$PC1, y=df_out$PC2)
      centroids <- aggregate(cbind(x,y)~cluster,data=gg,mean)
      gg <- merge(gg,centroids,by="cluster",suffixes=c("",".centroid"))
      
      ggplot(gg)+
      geom_point(aes(x=x,y=y,color=cluster, fill = cluster), size=3)+
      stat_ellipse(geom = "polygon", aes(x=x,y=y,color=cluster, fill = cluster), alpha = 0.3) +
      geom_point(data=centroids, aes(x=x, y=y, color=cluster, fill = cluster), size = 5, alpha = 0.9, shape = 19)+ 
      geom_segment(aes(x=x.centroid, y=y.centroid, xend=x, yend=y, color=cluster))+
      theme(legend.title=element_text(size=15),legend.text=element_text(size=12),
            legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
            axis.title.y=element_text(size=15),axis.title.x=element_text(size=15),
            axis.text.y=element_text(size=18),axis.text.x=element_text(size=18))+
      xlab(paste0("PC1: ",round(a1,0),"% variance"))+
      ylab(paste0("PC2: ",round(a2,0),"% variance"))+ 
        theme(axis.title.x = element_text(size=16, face = "bold",color="black"),
              axis.title.y = element_text(size=16, face = "bold",color="black"),
              axis.text = element_text(color="black",size=16, face = "bold"),
              axis.text.x =element_text(color="black",size=16, face = "bold"),
              panel.border = element_blank(),
              axis.ticks.x = element_blank(),
              strip.text.x = element_text(size = 9,color="black"),
              panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
              panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
              panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
              plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
              legend.background=element_rect(fill = alpha("white", 0)))+ scale_color_manual(values = col)+ scale_fill_manual(values = col)
      path_fig <- paste0("results/figures/PCA/", analysis, "_", type, "_", y, "_centroids.pdf")
      ggsave(path_fig, height = 6, width = 7)
      
      }else{
          ggplot(df_out, aes( x= PC1, y = PC2, color = group))+ geom_point(size = 5, alpha = 0.9, shape = 19)+ 
        theme(legend.title=element_text(size=15),legend.text=element_text(size=12),
              legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
              axis.title.y=element_text(size=15),axis.title.x=element_text(size=15),
              axis.text.y=element_text(size=18),axis.text.x=element_text(size=18))+
        xlab(paste0("PC1: ",round(a1,0),"% variance")) +
        ylab(paste0("PC2: ",round(a2,0),"% variance"))+ 
          theme(axis.title.x = element_text(size=16, face = "bold",color="black"),
                axis.title.y = element_text(size=16, face = "bold",color="black"),
                axis.text = element_text(color="black",size=16, face = "bold"),
                axis.text.x =element_text(color="black",size=16, face = "bold"),
                panel.border = element_blank(),
                axis.ticks.x = element_blank(),
                strip.text.x = element_text(size = 9,color="black"),
                panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
                panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
                panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
                plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
                legend.background=element_rect(fill = alpha("white", 0)))+ scale_color_manual(values = col)+ scale_fill_manual(values = col)
        
      }
      path_fig <- paste0("results/figures/PCA/", analysis, "_", type, "_", y, ".pdf")
      ggsave(path_fig, height = 6, width = 7)
    }
}
```

```{r}
library(dplyr)
```

```{r}
clinical_2 <- clinical[,c(1:2)]
data_olink$COCOMO_ID <- rownames(data_olink)
data <- merge(clinical_2, data_olink, by = "COCOMO_ID")


## PCA (without outliers)
pca <- prcomp(data[,-c(1:2)], scale. = FALSE)
df_out <- as.data.frame(pca$x)
df_out$group <- as.factor(data$cluster)
rownames(df_out) <- data$COCOMO_ID
x <- project.pca.proportionvariances <- ((pca$sdev^2) / (sum(pca$sdev^2)))*100
a1 <- x[1]
a2 <- x[2]
gg <- data.frame(cluster=factor(df_out$group), x=df_out$PC1, y=df_out$PC2)
rownames(gg) <- rownames(df_out)
centroids <- aggregate(cbind(x,y)~cluster,data=gg,mean)
gg <- merge(gg,centroids,by="cluster",suffixes=c("",".centroid"))
rownames(gg) <- rownames(df_out)
gg$label <- rownames(gg)

ggplot(gg)+
    geom_point(aes(x=x,y=y,color=cluster, fill = cluster), size=3)+
    stat_ellipse(geom = "polygon", aes(x=x,y=y,color=cluster, fill = cluster), alpha = 0.3) +
    geom_point(data=centroids, aes(x=x, y=y, color=cluster, fill = cluster), size = 5, alpha = 0.9, shape = 19)+ 
    geom_segment(aes(x=x.centroid, y=y.centroid, xend=x, yend=y, color=cluster))+
    theme(legend.title=element_text(size=15),legend.text=element_text(size=12),
            legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
            axis.title.y=element_text(size=15),axis.title.x=element_text(size=15),
            axis.text.y=element_text(size=18),axis.text.x=element_text(size=18))+
      xlab(paste0("PC1: ",round(a1,0),"% variance"))+
      ylab(paste0("PC2: ",round(a2,0),"% variance"))+ 
        theme(axis.title.x = element_text(size=16, face = "bold",color="black"),
              axis.title.y = element_text(size=16, face = "bold",color="black"),
              axis.text = element_text(color="black",size=16, face = "bold"),
              axis.text.x =element_text(color="black",size=16, face = "bold"),
              panel.border = element_blank(),
              axis.ticks.x = element_blank(),
              strip.text.x = element_text(size = 9,color="black"),
              panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
              panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
              panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
              plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
              legend.background=element_rect(fill = alpha("white", 0)))+ scale_color_manual(values = col)+ scale_fill_manual(values = col)+ geom_text(aes(x=x,y=y, label = label))
      path_fig <- paste0("results/figures/PCA/PCA_centroids_labelled.pdf")
      ggsave(path_fig, height = 6, width = 7)
      
      
## remove outliers 
gg <- gg[!gg$label %in% c("2026", "2005"),]

ggplot(gg)+
    geom_point(aes(x=x,y=y,color=cluster, fill = cluster), size=3)+
    stat_ellipse(geom = "polygon", aes(x=x,y=y,color=cluster, fill = cluster), alpha = 0.3) +
    geom_point(data=centroids, aes(x=x, y=y, color=cluster, fill = cluster), size = 5, alpha = 0.9, shape = 19)+ 
    geom_segment(aes(x=x.centroid, y=y.centroid, xend=x, yend=y, color=cluster))+
    theme(legend.title=element_text(size=15),legend.text=element_text(size=12),
            legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
            axis.title.y=element_text(size=15),axis.title.x=element_text(size=15),
            axis.text.y=element_text(size=18),axis.text.x=element_text(size=18))+
      xlab(paste0("PC1: ",round(a1,0),"% variance"))+
      ylab(paste0("PC2: ",round(a2,0),"% variance"))+ 
        theme(axis.title.x = element_text(size=16, face = "bold",color="black"),
              axis.title.y = element_text(size=16, face = "bold",color="black"),
              axis.text = element_text(color="black",size=16, face = "bold"),
              axis.text.x =element_text(color="black",size=16, face = "bold"),
              panel.border = element_blank(),
              axis.ticks.x = element_blank(),
              strip.text.x = element_text(size = 9,color="black"),
              panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
              panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
              panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
              plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
              legend.background=element_rect(fill = alpha("white", 0)))+ scale_color_manual(values = col)+ scale_fill_manual(values = col)
      path_fig <- paste0("results/figures/PCA/PCA_centroids_whitout_outliers.pdf")
      ggsave(path_fig, height = 6, width = 7)
      
      
## UMAP
library(umap)

prot.labels <- data$cluster
prot.umap = umap(data[,-c(1:2)])
prot.umap
  
data <- prot.umap$data
layout <- prot.umap$layout
data_umap <- cbind(data, layout)
x = ncol(data_umap)
y = ncol(data_umap)-1
print(x)
print(y)
colnames(data_umap)[y:x] <- c("UMAP1", "UMAP2")
data_umap <-data.frame(data_umap)
  
data_umap %>% 
  mutate(Condition = prot.labels) %>%
  ggplot(aes(UMAP1, UMAP2), color = Condition, fill = Condition)+ geom_point(size = 5, alpha = 1, shape = 21, color = "black", aes(fill = factor(Condition)))+ theme_minimal()+ 
      theme(legend.title=element_text(size=10),legend.text=element_text(size=10),
            legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
            axis.title.y=element_text(size=16, color = "black", face="bold"),axis.title.x=element_text(size=16, color = "black", face="bold"),
            axis.text.y=element_text(size=14, color = "black"),axis.text.x=element_text(size=14, color = "black"))+ scale_color_manual(values = col)+ scale_fill_manual(values = col)
  
  ggsave(paste0("results/figures/UMAP_clusters_proteomics.pdf"), width = 6, height = 4)
```

```{r}
my_pca(data_olink, clinical, "cluster", "yes", FALSE, "yes", "prot", "PCA")
```

```{r}
list_names <- c("metabolomics", "transcriptomics", "olink")
list_data <- list(data_met, data_txn, data_olink)
list_parameters <- names(clinical)[-1]
list_parameters <- list_parameters[1]

for (i in 1:length(list_data)) {
  data_table <- list_data[[i]]
  type_ana <- list_names[i]
  name_ana <- "All_data"
  for (j in 1:length(list_parameters)) {
    parameter <- list_parameters[j]
    my_pca(data_table, clinical, parameter, "yes", FALSE, "yes", type_ana, name_ana)
    my_pca(data_table, clinical, parameter, "no", FALSE, "yes", type_ana, name_ana)
  }
}
```


## compare clusters
```{r}
#c1 <- read.csv("/home/flomik/Desktop/Code-PHD/COCOMO_txn/data/3_layers/cluster_SNF_3_omics_3_clusters.csv", stringsAsFactors = FALSE)
#c2 <- read.csv("processing/cluster_SNF_transcriptomics_metabolomics_olink_3.csv", stringsAsFactors = FALSE)

<<<<<<< HEAD
#match_clust <- merge(c1, c2, by = "names")
#match_clust <- match_clust[order(match_clust$cluster.y),]
=======
match_clust <- merge(c1, c2, by = "names")
match_clust <- match_clust[order(match_clust$cluster.y),]
match_clust$X.y <- NULL
match_clust$X.x <- NULL
>>>>>>> 91baeba32f362b1664f636ffb055e39424ac5cb2
```


```{r}
#library(ComplexHeatmap)
#pdf("results/figures/overlap_clusters_3.pdf")
#ha = HeatmapAnnotation(layers_3 = as.factor(match_clust$cluster.x), txn_met = as.factor(match_clust$cluster.y))
#draw(ha)
#dev.off()
```

```{r}
names(match_clust)[2:3] <- c("cluster_paper_3", "cluster_paper_4")
write.csv(match_clust, "manuscript/match_clusters_HIV.csv")
```



## overlap olink / transcriptomics

```{r}
library(ggvenn)
library(mygene)
library(org.Hs.eg.db)
```

```{r}
x <- colnames(data_olink)
y <- colnames(data_txn)
y2 <- mapIds(org.Hs.eg.db, y, keytype="ENSEMBL", column="SYMBOL")
y2 <- as.vector(y2)

list_genes <- list(
  olink = x,
  transcriptomics = y2
)

pdf("results/figures/ven_genes_detected.pdf")
ggvenn(list_genes)
dev.off()
```

## load DGE
```{r}
c1_txn <- read.csv("/home/flomik/Desktop/Code-PHD/COCOMO_txn/results/Deseq2/model3_results_DGE_2_1_filt.csv")
c2_txn <- read.csv("/home/flomik/Desktop/Code-PHD/COCOMO_txn/results/Deseq2/model2_results_DGE_2_1_filt.csv")
```

```{r}
c1_prot <- read.delim("/home/flomik/Desktop/Code-PHD/COCOMO_txn/results/LIMMA/model_3_LIMMA_results_olink_with_HC_filt.csv", sep = " ")
c2_prot <- read.delim("/home/flomik/Desktop/Code-PHD/COCOMO_txn/results/LIMMA/model_2_LIMMA_results_olink_with_HC_filt.csv", sep = " ")
```

```{r}
x3 <- c1_txn
y3 <- c1_prot

x4_up <- x3$Gene_name[x3$LFC_1_2 > 0]
x4_down <- x3$Gene_name[x3$LFC_1_2 < 0]

y4_up <- y3$BIOCHEMICAL[y3$logFC > 0]
y4_down <- y3$BIOCHEMICAL[y3$logFC < 0]

list_genes <- list(
  olink = y4_up,
  transcriptomics = x4_up
)

pdf("results/figures/ven_genes_diff_clusters_up_model_3.pdf")
ggvenn(list_genes)
dev.off()

list_genes <- list(
  olink = y4_down,
  transcriptomics = x4_down
)

pdf("results/figures/ven_genes_diff_clusters_down_model_3.pdf")
ggvenn(list_genes)
dev.off()
```
```{r}
x3 <- c2_txn
y3 <- c2_prot

x4_up <- x3$Gene_name[x3$LFC_1_2 > 0]
x4_down <- x3$Gene_name[x3$LFC_1_2 < 0]

y4_up <- y3$BIOCHEMICAL[y3$logFC > 0]
y4_down <- y3$BIOCHEMICAL[y3$logFC < 0]

list_genes <- list(
  olink = y4_up,
  transcriptomics = x4_up
)

pdf("results/figures/ven_genes_diff_clusters_up_model_2.pdf")
ggvenn(list_genes)
dev.off()

list_genes <- list(
  olink = y4_down,
  transcriptomics = x4_down
)

pdf("results/figures/ven_genes_diff_clusters_down_model_2.pdf")
ggvenn(list_genes)
dev.off()
```


