---
title: "RF models 3 layers integration metabolome"
output: html_notebook
---

### clean environment
```{r}
rm(list=ls())
```

### set directory
```{r setup}
    knitr::opts_knit$set(root.dir = normalizePath("~/Desktop/Code-PHD/COCOMO_txn/")) 
```




```{r}
gns <- read.csv("data/brain_genes_clean_table.csv")
```


## load senescence
```{r}
sen_1 <- read.delim("data/senescence/csgene_human.txt")
sen_2 <- read.delim("data/senescence/SASP_reactome.tsv")
sen_2$Gene_name <- gsub(".* ", "", sen_2$MoleculeName)
sen_3 <- read.delim("data/senescence/signatures_senescence.csv", sep = ";")
```
## table senescence
```{r}
library(org.Hs.eg.db)
library(mygene)
```

```{r}
senescence <- data.frame(Symbol = c(as.vector(sen_1$GeneSymb), as.vector(sen_2$Gene_name), as.vector(sen_3$gene_symbol)))
#senescence$Gene_name <- mapIds(org.Hs.eg.db, as.vector(senescence$Symbol), keytype="SYMBOL", column="ENSEMBL")
senescence$CSGene <- ifelse(senescence$Symbol %in% sen_1$GeneSymb, "yes", "no")
senescence$SASP <- ifelse(senescence$Symbol %in% sen_2$Gene_name, "yes", "no")
senescence$CellAge <- ifelse(senescence$Symbol %in% sen_3$gene_symbol, "yes", "no")
write.csv(senescence, "processing/table_senescent_genes.csv")
```


```{r}
library(VennDiagram)
# Chart
venn.diagram(
  x = list(sen_1$GeneSymb, sen_2$Gene_name, sen_3$gene_name),
  category.names = c("CSGene" , "SASP" , "CellAge"),
  filename = 'results/figures/overlap_senescent_databases.pdf',
  output=TRUE,
  cat.default.pos = "text"
)
```

```{r}
length(sen_2$Gene_name[sen_2$Gene_name %in% c(sen_1$GeneSymb, sen_3$gene_symbol)])
```

```{r}
library(dplyr)
library(ComplexHeatmap)
library(piano)
library(limma)
library(xlsx)
```

```{r}
col <- c( "#8ea76b", "#6b8ea7",	"#a7846b")
```

### clinical paraolinkers
```{r}
clinical <- read.csv("processing/clinical_data_clean_with_clusters_and_categories.csv")
```

### olink
```{r}
data <- read.csv("processing/olink_data_new_ids.csv", row.names = 1)
data$Condition <- NULL
```

```{r}
data$COCOMO_ID <- rownames(data)
pca_data <- merge(clinical, data, by = "COCOMO_ID")
data_prot <- data.frame(cluster = pca_data$cluster, CD4_CD8 = pca_data$CDA_CD8_cat, VAT = pca_data$VAT_cat, MetS = pca_data$METS, 
                        pca_data[,-c(1:47)])

rownames(data_prot) <- pca_data$COCOMO_ID
```

```{r}
data_SASP <- data_prot[,colnames(data_prot) %in% c("cluster", sen_2$Gene_name)]
data_SASP$cluster <- as.vector(data_SASP$cluster)
data_SASP$cluster[data_SASP$cluster == 1] <- "HC-like"
data_SASP$cluster[data_SASP$cluster == 2] <- "At-risk"
```

```{r}
data_SASP <- data_SASP[data_SASP$cluster != "Ctrl",]
```


```{r}
col <- c("#6b8ea7",	"#a7846b")
```

```{r}
library(ggpubr)
```

```{r}
data_SASP$cluster <- factor(data_SASP$cluster, levels = c("HC-like", "At-risk"))
```
```{r}
my_comparisons <- list(c("HC-like", "At-risk"))
```

```{r}
list_x <- list()
x = 1

for (i in 2:ncol(data_SASP)) {
  table_1 <- data_SASP[,c(i, 1)]
  a <- ggplot(table_1, aes(x = cluster, y = table_1[,1], fill = cluster)) + 
    geom_boxplot(position=position_dodge(1), size = 0.1, alpha =0.8, outlier.shape = NA)+ 
    labs(x="Cluster", y = paste0(names(table_1)[1], ""))+ 
    theme(axis.title.x = element_blank(),
          axis.title.y = element_text(size=16, face = "bold",color="black"),
          axis.text = element_text(color="black",size=16, face = "bold"),
          axis.text.x =element_text(color="black",size=16, face = "bold"),
          panel.border = element_blank(),
          axis.ticks.x = element_blank(),
          strip.text.x = element_text(size = 9,color="black"),
          legend.position="none",
          panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
          panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
          panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
          plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
          legend.background=element_rect(fill = alpha("white", 0))) + 
    geom_jitter(shape=16, size=1, color="black", position=position_jitter(0.05))+ scale_fill_manual(values=col)+ 
  stat_compare_means(comparisons = my_comparisons)
  a
  ggsave(paste0("results/figures/boxplots_SASP/SASP_boxplot_", colnames(data_SASP)[i], ".pdf"), height = 4, width = 3.5)
  print(i)
  list_x[[x]] <- a
  x <- x + 1
}
```

```{r}
data_prot <- data_prot[data_prot$cluster != "Ctrl",]
```

```{r}
sen <- c(sen_3$gene_symbol, sen_1$gene_symbol, sen_2$Gene_name)
sen <- unique(sen)
```

```{r}
dge <- read.csv("results/LIMMA/model_3_LIMMA_results_olink_with_HC_filt.csv", sep = " ")
dge <- dge[dge$adj.P.Val < 0.05,]
```

```{r}
cluster <- data_prot$cluster
CD4_CD8 <- data_prot$CD4_CD8
VAT <- data_prot$VAT
MetS <- as.factor(data_prot$MetS)

data_prot_2 <- data.frame(t(data_prot[,-c(1:4)]))
data_prot_2$BIOCHEMICAL <- rownames(data_prot_2)
data_prot_2 <- merge(dge, data_prot_2, by = "BIOCHEMICAL")
```


```{r}
data_prot_3 <- data_prot_2[data_prot_2$BIOCHEMICAL %in% sen_1$GeneSymb,]
data_prot_4 <- data_prot_2[data_prot_2$BIOCHEMICAL %in% sen_2$Gene_name,]
data_prot_5 <- data_prot_2[data_prot_2$BIOCHEMICAL %in% sen_3$gene_symbol,]
```

```{r}
data_prot_6 <- data_prot_2[data_prot_2$BIOCHEMICAL %in% c(as.vector(data_prot_4$BIOCHEMICAL),as.vector(data_prot_5$BIOCHEMICAL), as.vector(data_prot_3$BIOCHEMICAL)),]
data_prot_6$CSGene <- ifelse(data_prot_6$BIOCHEMICAL %in% sen_1$GeneSymb, "yes", NA)
data_prot_6$CellAge <- ifelse(data_prot_6$BIOCHEMICAL %in% sen_3$gene_symbol, "yes", NA)
data_prot_6$SASAP <- ifelse(data_prot_6$BIOCHEMICAL %in% sen_2$Gene_name, "yes", NA)

table(data_prot_6$SASAP)
table(data_prot_6$CellAge)
table(data_prot_6$CSGene)

rownames(data_prot_6) <- data_prot_6$BIOCHEMICAL

colnames(data_prot_6) <- gsub("X", "", colnames(data_prot_6) )
```

```{r}
data_prot_7 <- data_prot_6
```

```{r}
x <- data_prot_7[,c(164:166)]
```


```{r}
data_prot_6[,-c(1:5,164:166)] <- data.frame(t(scale(t(data_prot_6[,-c(1:5,164:166)]))))
```

```{r}
library(viridis)
col <- viridis(256)
```

```{r}
length(data_prot_6$BIOCHEMICAL[data_prot_6$logFC > 0])
length(data_prot_6$BIOCHEMICAL[data_prot_6$logFC < 0])
```

```{r}
library(circlize)
library(viridis)
col_fun1 = colorRamp2(c(-8,-4, -1.5, 0, 1.5, 4, 8), c("#7F7F00","#B2B200" ,"#E5E500","white","#BF7FBF","#993299","#590059"))
col_fun1 = colorRamp2(c(-8,-4, -1.5, 0, 1.5, 4, 8), c(col[1], col[20],col[50],col[125],col[175],col[215],col[256]))
col_fun1 = colorRamp2(c(-3,-1.5, -0.5, 0, 0.5, 1.5, 3), c(col[1], col[20],col[50],col[125],col[175],col[215],col[256]))
col_fun_2 = colorRamp2(c(-1, 0, 1), c("blue", "white", "red"))
```

SASAP --> 81
SASAP detected --> 12

SASAP proteomics (all overlap with GSGene database)

CDC26
CDC27
CDKN1A
CDKN2D
CXCL8
IGFBP7
NFKB1

SASAP transcriptomics --> MAPK1

```{r}
length(data_prot_6$logFC[data_prot_6$logFC < 0])
```


```{r}
col <- c( "#8ea76b", "#6b8ea7",	"#a7846b")
```

```{r}
col <- c( "#005900", "#878782",	"#761212")
```

```{r}
nrow(data_prot_6)
```
```{r}
nrow(data_prot_6[,-c(1:5,164:166)])
```
```{r}
data_prot_6$Synaptic_proteins <- ifelse(data_prot_6$BIOCHEMICAL %in% gns$HumanName, "yes", NA)
```

```{r}
pdf("results/figures/Heatmap_proteomics_senescence_proteins_significant_model_2_with_samples_label.pdf", width = 7, height = 4)
ht = Heatmap(data_prot_6[,-c(1:5,164:167)], 
             #column_order = sort(colnames(t(data_prot_6[,-c(1:5)]))),  
             col = col_fun1, 
             show_column_names = FALSE, 
             show_row_names = FALSE,
             #column_split = cluster, 
             #row_split = data_2$under_group, 
             #column_title = c("Ctrl",1, 2), 
             #column_title_gp = gpar(fill = col), 
             border = FALSE, 
             column_gap = unit(2, "mm"), 
             right_annotation = rowAnnotation(LogFC = data_prot_6$logFC, CSGene = data_prot_6$CSGene, SASP = data_prot_6$SASAP, CellAge = data_prot_6$CellAge, Synaptic = data_prot_6$Synaptic_proteins,
                                              col = list(LogFC = col_fun_2, CSGene = c("yes" = "#408040"), SASP = c("yes" = "#6aa84f"), CellAge = c("yes" = "#804080"), Synaptic = c("yes" = "#8b74d4")), show_legend = TRUE),
             top_annotation = HeatmapAnnotation(cluster = cluster,
                                              col = list(cluster = c("Ctrl" = col[1], "1" = col[2], "2" = col[3])), show_legend = TRUE)
             )
draw(ht)
dev.off()

ht
```
```{r}
write.csv(data_prot_6[,c(1:5,164:167)], "processing/table_heatmap_SAP.csv")
```


```{r}
prot_SASP <- data_prot_6[data_prot_6$SASAP == "yes",]
prot_SASP <- prot_SASP[complete.cases(prot_SASP$BIOCHEMICAL),]
prot_SASP <- data.frame(cluster = cluster, t(prot_SASP[,-c(1:5,164:166)]))
```

```{r}
prot_SASP$cluster <- as.vector(prot_SASP$cluster)
prot_SASP$cluster <- ifelse(prot_SASP$cluster == 1, "HC-like", "At-risk")
prot_SASP$cluster <- factor(prot_SASP$cluster, levels = c("HC-like", "At-risk"))
```

```{r}
my_comparisons <- list(c("HC-like", "At-risk"))
```

```{r}
list_x <- list()
x = 1

for (i in 2:ncol(prot_SASP)) {
  table_1 <- prot_SASP[,c(i, 1)]
  a <- ggplot(table_1, aes(x = cluster, y = table_1[,1], fill = cluster)) + 
    geom_boxplot(position=position_dodge(1), size = 0.1, alpha =0.8, outlier.shape = NA)+ 
    labs(x="Cluster", y = paste0(names(table_1)[1], ""))+ 
    theme(axis.title.x = element_blank(),
          axis.title.y = element_text(size=16, face = "bold",color="black"),
          axis.text = element_text(color="black",size=16, face = "bold"),
          axis.text.x =element_text(color="black",size=16, face = "bold"),
          panel.border = element_blank(),
          axis.ticks.x = element_blank(),
          strip.text.x = element_text(size = 9,color="black"),
          legend.position="none",
          panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
          panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
          panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
          plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
          legend.background=element_rect(fill = alpha("white", 0))) + 
    geom_jitter(shape=21, size=2, color="black", position=position_jitter(0.05))+ scale_fill_manual(values=col[-1])+ 
  stat_compare_means(comparisons = my_comparisons)
  a
  ggsave(paste0("results/figures/boxplots_SASP/SASP_boxplot_", colnames(prot_SASP)[i], "_v2.pdf"), height = 4, width = 3.5)
  print(i)
  list_x[[x]] <- a
  x <- x + 1
}
```
## check subcellular localisation
```{r}
library(org.Hs.eg.db)
library(mygene)

list_prot <- rownames(data_prot_6)
list_prot_accession <- mapIds(org.Hs.eg.db, list_prot, keytype="SYMBOL", column="UNIPROT")
```

```{r}
write.csv(list_prot_accession, "processing/list_proteins_SASP_localisation.csv")
```

```{r}
#table_location <- GetSubcellular_location(list_prot_accession, directorypath = NULL)
```
```{r}
types <- read.delim("/home/flomik/Desktop/Code-PHD/COCOMO_txn/data/uniprot/uniprot-compressed_true_download_true_fields_id_2Cname_2Ccategory_2C-2023.01.11-14.33.10.25.tsv")
localisation <- read.delim("/home/flomik/Desktop/Code-PHD/COCOMO_txn/data/uniprot/uniprot-compressed_true_download_true_fields_accession_2Creviewed_2C-2023.01.11-14.28.18.44.tsv")

table_genes_location <- data.frame(Location = unique(types$Name))

for (i in 1:nrow(localisation)) {
  gene <- as.vector(localisation[i,1])
  table_genes_location[,i+1] <- NA
  colnames(table_genes_location)[i + 1] <- gene
  for (j in 1:nrow(table_genes_location)) {
    table_genes_location[j, i + 1] <- ifelse(grepl(table_genes_location[j, 1], localisation[i,9]), 1, 0)
  }
}

table_genes_location <- table_genes_location[rowSums(table_genes_location[,-1]) > 0,]
table_genes_location[,-1] <- table_genes_location[,colSums(table_genes_location[,-1]) > 0]

table_genes_location$Location <- as.vector(table_genes_location$Location)
table_genes_location$sum <- rowSums(table_genes_location[,-1])

table_results <- data.frame(Area = table_genes_location$Location, Nb = table_genes_location$sum)
colSums(table_genes_location[,-c(1, 142)])

write.csv(table_genes_location, "results/localisation_proteins_senescence.csv")
write.csv(table_results, "results/localisation_proteins_senescence_number_per_localisation.csv")
```

```{r}
table_genes_location <- data.frame(Location = unique(types$Gene.Ontologies))
table_genes_location$Location <- gsub(".*\\:", "", table_genes_location$Location)

for (i in 1:nrow(localisation)) {
  gene <- as.vector(localisation[i,1])
  table_genes_location[,i+1] <- NA
  colnames(table_genes_location)[i + 1] <- gene
  for (j in 1:nrow(table_genes_location)) {
    table_genes_location[j, i + 1] <- ifelse(grepl(table_genes_location[j, 1], localisation[i,10]), 1, 0)
  }
}

table_genes_location <- table_genes_location[rowSums(table_genes_location[,-1]) > 0,]
#table_genes_location[,-1] <- table_genes_location[,colSums(table_genes_location[,-1]) > 0]

table_genes_location$Location <- as.vector(table_genes_location$Location)
table_genes_location$sum <- rowSums(table_genes_location[,-1])

table_results <- data.frame(Area = table_genes_location$Location, Nb = table_genes_location$sum)
colSums(table_genes_location[,-c(1, 142)])

write.csv(table_genes_location, "results/localisation_proteins_senescence_GO.csv")
write.csv(table_results, "results/localisation_proteins_senescence_GO_number_per_localisation.csv")
```

## PCA genes
```{r}
pca_data <- data.frame(condition = condition, t(data_prot_6))
pca <- prcomp(pca_data[,-c(1, 14)])
df_out <- as.data.frame(pca$x)
df_out$group <- as.factor(pca_data$condition)
x <- project.pca.proportionvariances <- ((pca$sdev^2) / (sum(pca$sdev^2)))*100
a1 <- x[1]
a2 <- x[2]

gg <- data.frame(cluster=factor(df_out$group), x=df_out$PC1, y=df_out$PC2)
rownames(gg) <- rownames(df_out)
centroids <- aggregate(cbind(x,y)~cluster,data=gg,mean)
gg <- merge(gg,centroids,by="cluster",suffixes=c("",".centroid"))
rownames(gg) <- rownames(df_out)
gg$label <- rownames(gg)
```

```{r}
ggplot(df_out)+
    geom_point(aes(x=PC1,y=PC2,color=group, fill = group), size=3) +
    theme(legend.title=element_text(size=15),legend.text=element_text(size=12),
            legend.key.size=unit(0.7,"line"),plot.title = element_text(hjust = 0.5),
            axis.title.y=element_text(size=15),axis.title.x=element_text(size=15),
            axis.text.y=element_text(size=18),axis.text.x=element_text(size=18))+
      xlab(paste0("PC1: ",round(a1,0),"% variance"))+
      ylab(paste0("PC2: ",round(a2,0),"% variance"))+ 
        theme(axis.title.x = element_text(size=16, face = "bold",color="black"),
              axis.title.y = element_text(size=16, face = "bold",color="black"),
              axis.text = element_text(color="black",size=16, face = "bold"),
              axis.text.x =element_text(color="black",size=16, face = "bold"),
              panel.border = element_blank(),
              axis.ticks.x = element_blank(),
              strip.text.x = element_text(size = 9,color="black"),
              panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
              panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
              panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
              plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
              legend.background=element_rect(fill = alpha("white", 0)))+ scale_color_manual(values = col)+ scale_fill_manual(values = col)+ scale_fill_manual(values = col)+ geom_density_2d(aes(x=PC1,y=PC2,color=group, fill = group))


path_fig <- paste0("results/figures/PCA_olink_macrophages_filtered_29_style_4_density.pdf")
ggsave(path_fig, height = 4, width = 5)
```

extract 20 samples
```{r}
list_patients <- colnames(ht@matrix)[column_order(ht)]

n <- 8

list_patients_top <- c(list_patients[1:n],list_patients[151:158])

data_prot_7 <- data_prot_7[,colnames(data_prot_7) %in% list_patients_top]

data_prot_7 <- data.frame(t(data_prot_7))
```

```{r}
clinical <- clinical[,c(2,3)]

clinical_2 <- read.csv("/home/flomik/Desktop/Code-PHD/COCOMO_txn/data/clinical/data_base_200_patients_COCOMO_filtered.csv")
clinical_2 <- clinical_2[,c(1, 2, 4)]
names(clinical_2)[1] <- "COCOMO_ID"

clinical <- merge(clinical, clinical_2, by = "COCOMO_ID")
```

```{r}
data_prot_7$COCOMO_ID <- rownames(data_prot_7)

data_prot_8 <- merge(clinical[,c(1, 2, 3)], data_prot_7, by = "COCOMO_ID")
```



```{r}
data_prot_8 <- data_prot_8[, colnames(data_prot_8) %in% c("COCOMO_ID", "cluster", as.vector(sen_1$GeneSymb))]
```

```{r}
colnames(data_prot_8)[colnames(data_prot_8) %in% c("CXCL8")]
```

```{r}
data_prot_8$cluster <- as.factor(as.vector(data_prot_8$cluster))
```

```{r}
table(data_prot_8$cluster)
```
## limma


```{r}
group <- as.factor(data_prot_8$cluster)
#bmi <- pca_data$BMI
#gender <- as.factor(pca_data$GENDER)
#age <- pca_data$AGE

design <- model.matrix( ~ 0 + group)
```



```{r}
fit <- lmFit(t(data_prot_8[,-c(1:2)]), design)
```

```{r}
list_comp <- c(paste0("group",2:1))
list_pair <- data.frame(t(combn(list_comp,2)))
list_pair$comp <- paste0(list_pair$X1, "-", list_pair$X2)
comp <- list_pair$comp
names(comp) <- gsub("roup", "", comp)
```

```{r}
cont.matrix <- makeContrasts(G2_G1 = comp[1],
                             levels=design)

fit.cont <- contrasts.fit(fit, cont.matrix)
fit.cont <- eBayes(fit.cont)
```

```{r}
top_table <- data.frame(BIOCHEMICAL = NA, logFC= NA, P.Value= NA, adj.P.Val= NA, Comp= NA)

for (i in 1:length(comp)) {
  top_table_1 <- topTable(fit.cont, coef=i, adjust="BH", n = Inf, sort.by = "P")
  top_table_1$Comp <- comp[i]
  top_table_1$BIOCHEMICAL <- rownames(top_table_1)
  top_table_1 <- top_table_1[,c(8, 1, 4, 5, 7)]
  write.table(top_table_1$BIOCHEMICAL[top_table_1$adj.P.Val < 0.1], paste0("processing/MSEA/model_senescence_16_olink_", comp[i],".txt"), quote = FALSE, row.names = F)
  top_table <- rbind(top_table, top_table_1)
}

top_table <- top_table[complete.cases(top_table),]
top_table$sign <- ifelse(top_table$logFC > 0, "up in at-risk", "down in at-risk")
path_table <- paste0("results/LIMMA/model_senescence_30_LIMMA_results_olink.csv")

print(path_table)
write.csv(top_table, file =path_table)

DF_top_table <-top_table[top_table$adj.P.Val < 0.1, ]

path_results <- paste0("results/LIMMA/model_senescence_16_LIMMA_results_olink_filt.csv")

write.table(DF_top_table, file = path_results)

path_results <- paste0("results/LIMMA/model_senescence_16_LIMMA_results_olink_filt.xlsx")

write.xlsx(DF_top_table, file = path_results)

table(DF_top_table$Comp)
```
## boxplots
```{r}
data_prot <- data_prot_8
```

```{r}
data_prot$cluster <- as.vector(data_prot$cluster)
data_prot$cluster[data_prot$cluster == 1] <- "HC-like"
data_prot$cluster[data_prot$cluster == 2] <- "At-risk"

data_prot <- data_prot[order(data_prot$cluster),]
write.csv(data_prot[,c(1,2)], "processing/selection_top_16_extreme_samples_senescent_proteins.csv")

data_prot <- data_prot[, colnames(data_prot) %in% c("cluster", DF_top_table$BIOCHEMICAL)]
```

```{r}
my_comparisons <- list(c("HC-like", "At-risk"))
```

```{r}
data_prot$cluster <- factor(data_prot$cluster, levels = c("HC-like", "At-risk"))
```

```{r}
library(ggpubr)
```

```{r}
col <- col[-1]
```

```{r}
list_x <- list()
x = 1

for (i in 2:ncol(data_prot)) {
  table_1 <- data_prot[,c(i, 1)]
  a <- ggplot(table_1, aes(x = cluster, y = table_1[,1], fill = cluster)) + 
    geom_boxplot(position=position_dodge(1), size = 0.1, alpha =0.8, outlier.shape = NA)+ 
    labs(x="Cluster", y = paste0(names(table_1)[1], ""))+ 
    theme(axis.title.x = element_blank(),
          axis.title.y = element_text(size=16, face = "bold",color="black"),
          axis.text = element_text(color="black",size=16, face = "bold"),
          axis.text.x =element_text(color="black",size=16, face = "bold"),
          panel.border = element_blank(),
          axis.ticks.x = element_blank(),
          strip.text.x = element_text(size = 9,color="black"),
          legend.position="none",
          panel.background = element_rect(fill = "white", colour = "white", size = 0.5, linetype = "solid"),
          panel.grid.major = element_line(size = 0.5, linetype ='solid', colour = "grey"),
          panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "grey"),
          plot.title = element_text(hjust = 0.5, size = 20, face = "bold", vjust=2),
          legend.background=element_rect(fill = alpha("white", 0))) + 
    geom_jitter(shape=16, size=1, color="black", position=position_jitter(0.05))+ scale_fill_manual(values=col)+ 
  stat_compare_means(comparisons = my_comparisons)
  a
  ggsave(paste0("results/figures/boxplots/boxplots_Ale/senescence_prot_", colnames(data_prot)[i], ".pdf"), height = 4, width = 3)
  print(i)
  list_x[[x]] <- a
  x <- x + 1
}
```



